File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03076/p03076_num6_parsed.py
A = int(input())
B = int(input())
C = int(input())
D = int(input())
E = int(input())

# order in decreasing order of serving time
order = sorted([(A, "ABC Don"), (B, "ARC Curry"), (C, "AGC Pasta"), (D, "APC Ramen"), (E, "ATC Hanbagu")], reverse=True)

# time variables
time = 0
delivery_time = 0

# iterate over each dish in the order
for dish_time, dish_name in order:
    # calculate the time for this dish to be delivered
    required_time = ((time + 9) // 10) * 10 + dish_time

    # check if order can be placed at this time or if we need to wait for the previous dish to be delivered
    if required_time > delivery_time:
        delivery_time = required_time
    else:
        delivery_time += dish_time

    # update the current time
    time = delivery_time

# print the earliest possible time for the last dish to be delivered
print(delivery_time)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03076/p03076_num6_parsed.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    int A, B, C, D, E;
    cin >> A >> B >> C >> D >> E;

    int max_time = max({A, B, C, D, E});  // Find the maximum serving time
    int total_time = 0;

    // Place orders in reverse order, starting from the maximum serving time
    for (int i = max_time; i >= 0; i -= 10) {
        if (i == A) {
            total_time += A;  // Serve ABC Don
            A = 0;  // Dish delivered, reset time
        }
        if (i == B) {
            total_time += B;  // Serve ARC Curry
            B = 0;  // Dish delivered, reset time
        }
        if (i == C) {
            total_time += C;  // Serve AGC Pasta
            C = 0;  // Dish delivered, reset time
        }
        if (i == D) {
            total_time += D;  // Serve APC Ramen
            D = 0;  // Dish delivered, reset time
        }
        if (i == E) {
            total_time += E;  // Serve ATC Hanbagu
            E = 0;  // Dish delivered, reset time
        }
    }

    cout << total_time << endl;

    return 0;
}